library/sql_upgrade_fx.php

```diff
@@ -77,6 +77,25 @@ function columnHasType($tblname, $colname, $coltype)
     return (strcasecmp($row['Type'], $coltype) == 0);
 }
 
+/**
+ * Check if a Sql column has a certain type and a certain default value.
+ *
+ * @param  string $tblname      Sql Table Name
+ * @param  string $colname      Sql Column Name
+ * @param  string $coltype      Sql Column Type
+ * @param  string $coldefault   Sql Column Default
+ * @return boolean              returns true if the sql column is of the specified type and default
+ */
+function columnHasTypeDefault($tblname, $colname, $coltype, $coldefault)
+{
+    $row = sqlQuery("SHOW COLUMNS FROM $tblname LIKE '$colname'");
+    if (empty($row)) {
+        return true;
+    }
+
+    return ((strcasecmp($row['Type'], $coltype) == 0) && (strcasecmp($row['Default'], $coldefault) == 0));
+}
```

Comment:
Not yet working, but I think I know why. The NULL is throwing things off. Will put some more thought into this tomorrow.